client/components/LearnerViewDirective.js

```diff
@@ -59,10 +59,9 @@ tie.directive('learnerView', [function() {
             <div class="tie-coding-window">
               <div class="tie-lang-terminal">
                 <div class="tie-coding-terminal">
-                  <ui-codemirror ui-codemirror="codeMirrorOptions"
+                  <ui-codemirror ui-codemirror-opts="codeMirrorOptions"
                       ng-model="code"
-                      class="tie-codemirror-container">
-                  </ui-codemirror>
+                      class="tie-codemirror-container"></ui-codemirror>
```

Comment:
Take a look at the difference between these 2 images. The first one is with the original version with the closing tag in a new line. The second one is with the change and the tag moved up.
![image](https://cloud.githubusercontent.com/assets/1597618/24731517/82b95e16-1a1f-11e7-9f4b-61381808258d.png)
![image](https://cloud.githubusercontent.com/assets/1597618/24731555/c5348d4c-1a1f-11e7-9712-6e190fc4a7e0.png)

As mentioned in the comment above, the blank line and space characters just before </ui-codemirror> are preserved in the editor. After undo is used, they appear in the editor.